### PR TITLE
Adding missing fields in aerospike-cluster-cr.yaml

### PR DESCRIPTION
**Pull request by tanmayja**

### CODE REVIEW

File: helm-charts/aerospike-cluster/templates/aerospike-cluster-cr.yaml [hunk 1]
```diff
@@ -13,11 +13,15 @@ metadata:
 
 spec:
   # Aerospike cluster size
-  size: {{ .Values.replicas | default 3 }}
+  size: {{ .Values.replicas }}
 
   # Aerospike server docker image
   image: {{ .Values.image.repository | default "aerospike/aerospike-server-enterprise" }}:{{ .Values.image.tag | default "7.1.0.0" }}
 
+  maxUnavailable: {{ .Values.maxUnavailable }}
```
Comment (abhishekdwivedi3060) [inline helm-charts/aerospike-cluster/templates/aerospike-cluster-cr.yaml:21]: We should add all these new fields in values.yaml for visibility

**Review by abhishekdwivedi3060 — CHANGES_REQUESTED** [review]

File: helm-charts/aerospike-cluster/templates/aerospike-cluster-cr.yaml [hunk 2]
```diff
@@ -106,3 +112,11 @@ spec:
...
   seedsFinderServices: {{- toYaml . | nindent 4 }}
   {{- end }}
+
+  {{- if .Values.rosterNodeBlockList }}
+  rosterNodeBlockList: {{ .Values.rosterNodeBlockList }}
+  {{- end }}
+
+  {{- if .Values.k8sNodeBlockList }}
+  k8sNodeBlockList: {{ .Values.k8sNodeBlockList }}
+  {{- end }}
```
Comment (abhishekdwivedi3060) [inline helm-charts/aerospike-cluster/templates/aerospike-cluster-cr.yaml:122]: This won't give the correct output, use something like
```
  {{- with .Values.k8sNodeBlockList }}
  k8sNodeBlockList:
  {{- toYaml . | nindent 4 }}
  {{- end }}
```

File: helm-charts/aerospike-cluster/values.yaml [hunk 3]
```diff
@@ -113,3 +113,23 @@ operatorClientCert: {}
...
+
+## rosterNodeBlockList is a list of blocked nodeIDs from roster in a strong-consistency setup
+## Replace the value with aerospike node id which needs to be blocked.
+rosterNodeBlockList:
+  - <aerospike-node-id>
+
+## k8sNodeBlockList is a list of Kubernetes nodes which are not used for Aerospike pods.
+## Replace the value with kubernetes cluster node name which needs to be blocked.
+k8sNodeBlockList:
+  - <node-name>
```
Comment (abhishekdwivedi3060) [inline helm-charts/aerospike-cluster/values.yaml:135]:
```suggestion
rosterNodeBlockList: []
#  - <aerospike-node-id>

## k8sNodeBlockList is a list of Kubernetes nodes which are not used for Aerospike pods.
## Replace the value with kubernetes cluster node name which needs to be blocked.
k8sNodeBlockList: []
#  - <node-name>
```

File: helm-charts/aerospike-cluster/templates/aerospike-cluster-cr.yaml (hunk 1 above)
Comment (abhishekdwivedi3060) [inline helm-charts/aerospike-cluster/templates/aerospike-cluster-cr.yaml:21]: Add comments for all the new fields added in this file

**Review by abhishekdwivedi3060 — APPROVED** [review]

**Review by sud82 — APPROVED** [review]

**Pull request merged**